tests/loops/epoch/test_training_epoch_loop.py

```diff
@@ -141,3 +143,17 @@ def test_prepare_outputs_training_batch_end_manual(batch_end_outputs, expected):
         num_optimizers=-1,  # does not matter for manual optimization
     )
     assert prepared == expected
+
+
+def test_no_val_on_train_epoch_loop_restart(tmpdir):
```

Comment:
Although I usually suggest unit-tests, can you write more of an integration test here?

The solution you propose here works but it's not ideal, it should get properly fixed with #8578: particularly this commit https://github.com/PyTorchLightning/pytorch-lightning/pull/8578/commits/e60215c05d2ae67a730d9126322d62fb1978e457

Since that branch will not need to change `_should_check_val_fx`, this test will become stale, so I would rather have a broader test which works for both PRs.

Thank you!

Reply:
if it's not ideal, then I'll make this draft for now and see once #8578 get's merged, if this issue still exists.
Since there are no more patch releases, are you planning to release it in v1.6?

Reply:
If it works for the bug-fix release, I'm totally fine with fixing it this way. #8578 cannot be applied to the bug-fix branch.

Also the extra test coverage is welcome

Reply:
since there are no more bugfix releases, will the patch in this PR still exists after your PR gets merged?

Reply:
In my PR I would remove the change you are doing here to `_should_check_val_fx`, but as I said, the test you add here will be helpful because it's something not tested in master.

> since there are no more bugfix releases

Where did you get this from? If there are bugs to fix and we can do the fix to the bug-fix release, we should. This would be one example

Reply:
> Where did you get this from?

since there wasn't any bugfix PR, so I assumed.

will update the test then :)